pkg/genericclioptions/genericclioptions.go

```diff
@@ -23,13 +28,120 @@ type IOStreams struct {
 	ErrOut io.Writer
 }
 
-type ClientConfig struct {
-	Kubeconfig    string
+type sharedClientConfig struct {
 	QPS           float32
 	Burst         int
 	UserAgentName string
-	RestConfig    *restclient.Config
-	ProtoConfig   *restclient.Config
+}
+
+func newSharedClientConfig(userAgentName string) sharedClientConfig {
+	return sharedClientConfig{
+		QPS:           50,
+		Burst:         75,
+		UserAgentName: userAgentName,
+	}
+}
+
+func (scc *sharedClientConfig) AddFlags(cmd *cobra.Command) {
+	cmd.PersistentFlags().Float32VarP(&scc.QPS, "qps", "", scc.QPS, "Maximum allowed number of queries per second.")
+	cmd.PersistentFlags().IntVarP(&scc.Burst, "burst", "", scc.Burst, "Allows extra queries to accumulate when a client is exceeding its rate.")
+}
+
+func (scc *sharedClientConfig) Validate() error {
+	return nil
+}
+
+func (scc *sharedClientConfig) Complete() error {
+	return nil
+}
+
+type ClientConfigSet struct {
+	sharedClientConfig
```

Comment:
QPS (an other options) can differ for clusters, e.g. the first one can run extra operator loops with automated multidc

Reply:
how do you make it configurable per cluster without changing the existing set of flags? Do you just allow for an array and assume the indexes are corresponding to kubeconfigs?

Reply:
This struct isn't specific to test framework, I only ask for the struct to allow different settings at this point.

Reply:
I see your point, but I think it does now? ClientConfigSet has a slice of ClientConfigs and each ClientConfig has configurable options (because it embeds ClientConfigBase) - so the ClientConfigBase allows to retain the flags as they are, but it could easily be removed from ClientConfigSet later on and configured per-client (or per-cluster)?

Reply:
@tnozicka ping

Reply:
I think that technically works, but a similar point about flag-only options being private would apply here as well